src/cli/init.rs

```diff
@@ -176,14 +293,170 @@ fn get_dir(path: PathBuf) -> Result<PathBuf, Error> {
     }
 }
 
+type PipReq = (PackageName, PyPiRequirement);
+type ParsedDependencies = (Vec<MatchSpec>, Vec<PipReq>, Vec<Arc<Channel>>);
+
+fn parse_dependencies(deps: Vec<CondaEnvDep>) -> miette::Result<ParsedDependencies> {
+    let mut conda_deps = vec![];
+    let mut pip_deps = vec![];
+    let mut picked_up_channels = vec![];
+    for dep in deps {
+        match dep {
+            CondaEnvDep::Conda(d) => {
+                let match_spec = MatchSpec::from_str(&d).into_diagnostic()?;
+                if let Some(channel) = match_spec.clone().channel {
+                    picked_up_channels.push(channel);
+                }
+                conda_deps.push(match_spec);
+            }
+            CondaEnvDep::Pip { pip } => pip_deps.extend(
+                pip.into_iter()
+                    .map(|mut dep| {
+                        let re = Regex::new(r"/([^/]+)\.git").unwrap();
+                        if let Some(caps) = re.captures(dep.as_str()) {
+                            let name= caps.get(1).unwrap().as_str().to_string();
+                            tracing::warn!("The dependency '{}' is a git repository, as that is not available in pixi we'll try to install it as a package with the name: {}", dep, name);
+                            dep = name;
+                        }
+                        let req = pep508_rs::Requirement::from_str(&dep).into_diagnostic()?;
+                        let name = rip::types::PackageName::from_str(req.name.as_str())?;
+                        let requirement = PyPiRequirement::from(req);
+                        Ok((name, requirement))
+                    })
+                    .collect::<miette::Result<Vec<_>>>()?,
+            ),
+        }
+    }
+
+    if !pip_deps.is_empty() {
+        conda_deps.push(MatchSpec::from_str("pip").into_diagnostic()?);
+    }
+
+    Ok((conda_deps, pip_deps, picked_up_channels))
+}
+
+fn parse_channels(channels: Vec<String>) -> Vec<String> {
+    let mut new_channels = vec![];
+    for channel in channels {
+        if channel == "defaults" {
+            // https://docs.anaconda.com/free/working-with-conda/reference/default-repositories/#active-default-channels
+            new_channels.push("main".to_string());
+            new_channels.push("r".to_string());
+            new_channels.push("msys2".to_string());
+        } else {
+            let channel = channel.trim();
+            if !channel.is_empty() {
+                new_channels.push(channel.to_string());
+            }
+        }
+    }
+    new_channels
+}
+
+fn read_env_yml(path: PathBuf) -> miette::Result<CondaEnvFile> {
```

Comment:
I think you should move this type into another file and implement `CondaEnvFile::from_path`.

Reply:
Did that